### PR TITLE
Feat: server error - 400 errors / other exceptions during /token

### PR DESCRIPTION
**Pull request by angela-tran**
Part of #2032 

 - [x] 400-level error or a non-`HTTPError` exception during `/token` returns a JSON response that is used to send user to `/error`
 - [x] `/error` shows server error template
 - [x] Sentry notification is sent
 - [x] Analytic event is sent (`failed access token request`)

### Reviewing / testing

**To test handling of 400-level HTTP errors**, you'll have to force it like in other PRs such as #2089.
To `/token`, add:
```diff
        try:
+          class response:
+             def __init__(self):
+                self.status_code = 400
+
+          raise HTTPError(response=response())
           response = client.request_card_tokenization_access()
        except Exception as e:
```

**To test other exceptions**:
- setting an incorrect `client_id` on `PaymentProcessor` causes `UnsupportedTokenTypeError` on `client.request_card_tokenization_access`
- setting an incorrect `api_base_url` on `PaymentProcessor` causes a `ConnectionError` on `client.ensure_active_token`
- setting an incorrect `client_secret_name` on `PaymentProcessor` could cause an error when creating `Client` (we read from `payment_processor.client_secret_name`)

### CODE REVIEW

File: benefits/enrollment/views.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
 ROUTE_INDEX = "enrollment:index"
 ROUTE_REENROLLMENT_ERROR = "enrollment:reenrollment-error"
 ROUTE_RETRY = "enrollment:retry"
+ROUTE_SERVER_ERROR = "core:server-error"
```
Comment (thekaveman) [inline benefits/enrollment/views.py:24]: Does it feel weird to have this definition here? Should it be imported from `core.urls` or something?

File: benefits/enrollment/templates/enrollment/index.html [hunk 2]
```diff
@@ -45,6 +45,7 @@ <h1 class="pb-lg-8 pb-4">
                 // https://stackoverflow.com/a/42469170
                 // use 'assign' because 'replace' was giving strange Back button behavior
                 window.location.assign(data.redirect);
+                return;
```
Comment (thekaveman) [inline benefits/enrollment/templates/enrollment/index.html:48]: Why is this needed? I thought it was working in #2089? Maybe add a small comment

Comment (thekaveman) [reply benefits/enrollment/templates/enrollment/index.html:48]: Ah, I see the commit message now. I think a code comment would help thought!

File: benefits/enrollment/views.py [hunk 3]
```diff
@@ -38,25 +38,35 @@ def token(request):
...
             sentry_sdk.capture_exception(e)
 
-            if isinstance(e, HTTPError) and e.response.status_code >= 500:
-                analytics.failed_access_token_request(request, e.response.status_code)
-                data = {"redirect": reverse(ROUTE_SYSTEM_ERROR)}
-                return JsonResponse(data)
+            if isinstance(e, HTTPError):
+                status_code = e.response.status_code
+
+                if e.response.status_code >= 500:
```
Comment (thekaveman) [inline benefits/enrollment/views.py:58]: Small suggestion to use the `status_code` variable here instead

**Review by thekaveman — CHANGES_REQUESTED** [review]
Very minor request, otherwise this looks good!

File: benefits/enrollment/views.py (hunk 3 above)
Comment (angela-tran) [reply benefits/enrollment/views.py]: Thanks for catching. Done in [fc291ed](https://github.com/cal-itp/benefits/pull/2117/commits/fc291ed866e52c07681a64bd81977554e0188f96)

**Review by thekaveman — APPROVED** [review]
:+1:
